velodyne_pointcloud/src/conversions/convert.cc

```diff
@@ -43,19 +45,50 @@ namespace velodyne_pointcloud
         ROS_ERROR_STREAM("Could not load calibration file!");
     }
 
-    if(config_.organize_cloud)
+    // container selection and check
+    std::string container_name;
+    container_configured_ = private_nh.getParam("container", container_name);
+    private_nh.param<bool>("organize_cloud", config_.organize_cloud, false);
+    if(!container_configured_)
     {
-      container_ptr_ = boost::shared_ptr<OrganizedCloudXYZIR>(
-          new OrganizedCloudXYZIR(config_.max_range, config_.min_range,
-              config_.target_frame, config_.fixed_frame,
-              config_.num_lasers, data_->scansPerPacket()));
+      std::string default_container_name, default_organized_container_name;
+      std::stringstream container_list;
+      for (const auto& pair : container_names_)
+      {
+        container_list << pair.first << ", ";
+      }
+
+      if(config_.organize_cloud)
+      {
+        ROS_WARN_STREAM("The parameter \"organize_cloud\" is deprecated, please use the \"container\" param "
+                        "with one of the organized containers!");
+        container_name = default_organized_container_.first;
+        container_ptr_ = getContainer(default_organized_container_.second);
+      }
+      else
+      {
+        container_name = default_container_.first;
+        container_ptr_ = getContainer(default_container_.second);
+      }
+      ROS_WARN_STREAM("Using default container \"" << container_name << "\"!");
+      ROS_WARN_STREAM("Please use the \"container\" param with one of the following container names: " << container_list.str());
```

Comment:
Please either do one of the following:

1. Add the `container` parameter to all launch files and set it to the un-organized default of `PointCloudXYZIR`; or
2. Set this to be an `INFO` statement instead of a warning.

Having the new default value cause a warning seems problematic and could unnecessarily alarm users. I would prefer the first option since it should also be accompanied by some documentation with the possible values in the launch file.

Reply:
I also prefer the first option! Will do that.